mp/src/game/client/prediction.cpp

```diff
@@ -882,19 +869,6 @@ void CPrediction::RunCommand( C_BasePlayer *player, CUserCmd *ucmd, IMoveHelper
 		}
 	}
 
-	// Latch in impulse.
-	IClientVehicle *pVehicle = player->GetVehicle();
-	if ( ucmd->impulse )
-	{
-		// Discard impulse commands unless the vehicle allows them.
-		// FIXME: UsingStandardWeapons seems like a bad filter for this. 
-		// The flashlight is an impulse command, for example.
-		if ( !pVehicle || player->UsingStandardWeaponsInVehicle() )
-		{
-			player->m_nImpulse = ucmd->impulse;
-		}
-	}
```

Comment:
Bad removal. This was checking if the vehicle didn't exist, so we still need this.